R/maths_correctForMemoryEffect.R

```diff
@@ -46,6 +46,30 @@ calculateMemoryCoefficients <- function(dataset) {
   # join the mean mem coeff and the mem coeff for each standard into one dataframe
   tablesToJoin <- append(memoryCoeffForEachStandard, list(meanMemoryCoefficients))
   memCoeffOutput <- reduce(tablesToJoin, full_join, by = "Inj Nr")
+
+  # remove all trailing rows from output dataframe which only contain NA values
+  memCoeffOutput <- memCoeffOutput %>%
+    filter(cumall(!is.na(memoryCoeffD18O)))
+
+  # pad mean memory coefficients with 1's if last coefficient refers to an
+  # injection number less than the maximum inj. number of the samples/standards
+  # in the other blocks
```

Comment:
I would like to suggest an alternative solution to this. Instead of padding with 1 here, we could modify the function `applyMemoryCorrection` that uses the memory coefficients calculated here. 

The result could look as follows
```R
# function applyMemoryCorrection
...

# use a left join here to keep all rows from the sample data
joinedData <- left_join(sampleData, memoryCoefficients, by = c("Inj Nr"))

# set the memory coeff to 1 for all lines that have an injection number greater than the maximum injection number in the memory coefficients
joinedData[joinedData$`Inj Nr` > max(memoryCoefficients$`Inj Nr`), c("memoryCoeffD18O", "memoryCoeffDD")] <- 1
```
By doing it as suggested here, we would only need to add one line of code.

Please let me know if this works. Note that I have also added some comments to your code below.

Reply:
Thanks for this neat solution! This is superior to my approach since it is less additional code and, more importantly, it does not change the output of the estimated memory coefficients, so there is no need to adapt anything in Cpt-Picarr.

I changed the code to use this approach and updated the respective tests accordingly, see 
248cc8e and b25a440.